### PR TITLE
#12 - create docs about Vue

### LINKED ISSUE #12: Create doc about Vue

### PR DESCRIPTION
**Pull request by kvas-damian**
Fixes #12 

Ping @swacisko @Niko998 - let me know if it's useful or something is missing.

### CODE REVIEW

File: vue.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+* I can dynamically switch between components using `<component>`
+* I can use filters for data transformations in templates
+
+### Advanced aka _TODO SPA_
+* I can use an external plugin
+* I can use routes to navigate in the application using Vue router
+* I can use Vuex state manager to simplify data flow in the application
+* I can validate Prop Types
+* I can use scoped slots to access data from the child component
+* I can use `<transition>` component to make the application more smooth
```
Comment (bognix) [inline vue.md]:
```suggestion
* I can use `<transition>` component to make the application smoother
```

File: vue.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+* I can prepare production build
+
```
Comment (bognix) [inline vue.md:42]: I would mention somewhere mixins

File: vue.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+### Apprentice aka _Hello World_
+* I can instantiate Vue object
+* I can create a component
+* I understand Vue component lifecycle
+* I can use Vue template syntax to render component HTML
+
+### Intermediate aka _My first SPA_
+**Prerequisite knowledge**: sass
+* I can use conditional rendering (`v-if` vs `v-show`)
```
Comment (bognix) [inline vue.md:18]: I would mention different ways of registering components:
- global components
- import component file inside different component

File: vue.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+* I can use Vuex state manager to simplify data flow in the application
+* I can validate Prop Types
+* I can use scoped slots to access data from the child component
+* I can use `<transition>` component to make the application more smooth
+* I can prepare production build
+
+### Master aka _Under the hood_
+**Prerequisite knowledge**: debugger
+* I can create my own plugin
+* I can debug Vue application using Vue.js devtools
```
Comment (bognix) [inline vue.md]: Vue devtools can be useful right from the beginning. What is more, I think they can make it easier to understand what's happening under the hood.

File: vue.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+* I can use Vue template syntax to render component HTML
+
+### Intermediate aka _My first SPA_
+**Prerequisite knowledge**: sass
+* I can use conditional rendering (`v-if` vs `v-show`)
+* I can render a list (`v-for`)
+* I can style a component
+* I can pass pass parameters to a component (`props`)
+* I can emit and handle component events (`$emit`)
+* I can use computed properties and watchers
```
Comment (bognix) [inline vue.md:23]: I would mention somewhere the difference between computed and data.

File: vue.md (hunk 4 above)
Comment (kvas-damian) [reply vue.md]: I personally wasn't using such tools at low levels. What is your experience?

File: vue.md (hunk 3 above)
Comment (kvas-damian) [reply vue.md:18]: > import component file inside different component

IMO this is a default way to import components. I'll mention about global components somewhere.

File: vue.md (hunk 5 above)
Comment (kvas-damian) [reply vue.md:23]: Done

File: vue.md (hunk 2 above)
Comment (kvas-damian) [reply vue.md:42]: Done

File: vue.md (hunk 4 above)
Comment (bognix) [reply vue.md]: I started using it right away. I'm using dev tools for other frameworks as well and at the very beginning I use them to track the data flow and better understand what is happening under the hood. But this is my default way of learning, first understand, then write :)

**Pull request merged**